### PR TITLE
maturin: migrate to python@3.10

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by messense** [comment]
The failure is odd, we could try to change 

```ruby
system "python3", "-m", "pip", "install", "hello_world", "--no-index", "--find-links", testpath/"dist"
```

to 

```ruby
system "python3", "-m", "pip", "install",  testpath/"dist/hello_world*.whl"
```

**Comment by carlocab** [comment]
The test probably fails because it's using the CLT-provided `python3` in the `test` block. `python@3.10` is keg-only, so won't be in `PATH`.

**Comment by cclauss** [comment]
What is the remaining TODO on this PR given that this passes all tests and [maturin v0.12.0](https://github.com/PyO3/maturin/releases/tag/v0.12.0) supports Python 3.10?

File: Formula/maturin.rb [hunk 1]
```diff
@@ -32,7 +32,8 @@ def install
   test do
```
Comment (carlocab) [inline Formula/maturin.rb:32]:
```suggestion
  test do
    ENV.prepend_path "PATH", Formula["python@3.10"].opt_bin
```

File: Formula/maturin.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ def install
...
     system "cargo", "new", "hello_world", "--bin"
     system bin/"maturin", "build", "-m", "hello_world/Cargo.toml", "-b", "bin", "-o", "dist", "--compatibility", "off"
-    system "python3", "-m", "pip", "install", "hello_world", "--no-index", "--find-links", testpath/"dist"
-    system "python3", "-m", "pip", "uninstall", "-y", "hello_world"
+    system Formula["python@3.10"].opt_bin/"python3", "-m", "pip", "install", "hello_world", "--no-index",
+    "--find-links", testpath/"dist"
+    system Formula["python@3.10"].opt_bin/"python3", "-m", "pip", "uninstall", "-y", "hello_world"
```
Comment (carlocab) [inline Formula/maturin.rb:37]:
```suggestion
    system "python3", "-m", "pip", "install", "hello_world", "--no-index", "--find-links", testpath/"dist"
    system "python3", "-m", "pip", "uninstall", "-y", "hello_world"
```

**Review by carlocab — COMMENTED** [review]
We don't need to hardcode `python@3.10` in multiple places that will need changing eventually.

**Review by carlocab — APPROVED** [review]

**Pull request merged**